File: server_integration_tests/servers/java_server_reactive/src/main/java/dev/graphlink/javaserver/impl/ArticleSchemaMappingsServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.services.ArticleSchemaMappingsService;
import dev.graphlink.javaserver.generated.services.ArticleWithCountSchemaMappingsService;
import dev.graphlink.javaserver.generated.types.Article;
import dev.graphlink.javaserver.generated.types.Author;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.stream.Collectors;

@Service
public class ArticleSchemaMappingsServiceImpl
        implements ArticleSchemaMappingsService, ArticleWithCountSchemaMappingsService {

    @Override
    public Mono<Author> articleAuthor(Article value) {
        return Mono.justOrEmpty(Data.authors.stream()
                .filter(a -> a.getId().equals(value.getAuthorId()))
                .findFirst()
                .orElse(null));
    }

    @Override
    public Flux<Author> articleAuthorList(Article value) {
        return Flux.fromIterable(Data.authors.stream()
            .filter(a -> a.getId().equals(value.getAuthorId()))
            .collect(Collectors.toList()));
    }

    @Override
    public Mono<Integer> articleWithCountCount(Article value) {
        return Mono.just(Data.articles.indexOf(value) + 1);
    }
}

File: server_integration_tests/kotlin/kotlin_server_suspend/src/main/kotlin/dev/graphlink/kotlinserversuspend/impl/ArticleSchemaMappingsServiceImpl.kt
package dev.graphlink.kotlinserversuspend.impl

import dev.graphlink.kotlinserversuspend.generated.services.ArticleSchemaMappingsService
import dev.graphlink.kotlinserversuspend.generated.services.ArticleWithCountSchemaMappingsService
import dev.graphlink.kotlinserversuspend.generated.types.Article
import dev.graphlink.kotlinserversuspend.generated.types.Author
import org.springframework.stereotype.Service

@Service
class ArticleSchemaMappingsServiceImpl :
    ArticleSchemaMappingsService, ArticleWithCountSchemaMappingsService {

    override suspend fun articleAuthor(value: Article): Author =
        Data.authors.first { it.id == value.authorId }

    override suspend fun articleAuthorList(value: Article): List<Author>? =
        Data.authors.filter { it.id == value.authorId }

    override suspend fun articleWithCountCount(value: Article): Int = Data.articles.indexOf(value) + 1
}
